### ISSUE TITLE
Issue with Ajax POST calls on page loads

### ISSUE THREAD

**Issue by bhavinrshah** [mebjas/CSRF-Protector-PHP#113]
JS Request:
    <script>
        $(document).ready(function(){
           // ajax call
          var postUrl  = 'getData.php';
          var items = [];
          var item = {
            strReleaseName      : 'test'
          };
          items.push(item);
          var sortorder = {
            items : items
          };
          $.ajax({
            type : 'POST',
            url : postUrl,
            data :  "format=json&data="+ JSON.stringify(sortorder),
            dataType: "json",
            success : function(data, textStatus, XMLHttpRequest){
              alert('HI');
            },
            error: function (xhr, textStatus, errorThrown){
                alert('Error');
            }
          });
        });
    </script>

Request Header:

![CSRF Error](https://user-images.githubusercontent.com/3405847/57864700-fd423180-7819-11e9-9b80-357f57b418fa.jpg)

Please check.

**Comment by bhavinrshah** [comment]
@mebjas : Can you please look into this?

**Comment by mebjas** [comment]
@bhavinrshah on it

**Comment by mebjas** [comment]
So it looks to be related to this - https://github.com/mebjas/CSRF-Protector-PHP/pull/112

As a temp hack you could try making changes as suggested in this PR.

**Comment by bhavinrshah** [comment]
I tried but that is not working.
I have to replace `$(document).ready(function() {` to `$(window).bind('load', function() {` and it works. But still if it can be done through `csrfprotector.js` that would be great.

**Comment by mebjas** [comment]
Yes please make the required changes as dont in that PR in CSRF Protector code. Can you have the projec ton github? I'll send a PR to the same to test this scenario further.

@bhavinrshah

**Comment by bhavinrshah** [comment]
@mebjas : Let me create a test project on github and let you know

**Comment by bhavinrshah** [comment]
@mebjas : Here is link https://github.com/bhavinrshah/csrf_issue

**Comment by mebjas** [comment]
@bhavinrshah will setup the stack and check. Stay tuned.

**Comment by bhavinrshah** [comment]
@mebjas: Do we have any update on this?

**Comment by mebjas** [comment]
There is a missing submit.php file to completely reproduce this demo. Can you add some mock submit.php so I can test end to end?
@bhavinrshah
